### PR TITLE
Uses real rate engine for weight slider [delivers #156899536]

### PR DESCRIPTION
**Pull request by stangah**
## Description

Uses real rate engine API to inform incentive estimates on weight slider page. Also updates rate engine to prorate weights under 1000lbs, as done in [move.mil](https://github.com/deptofdefense/move.mil/blob/6af44afedc1308110c2a722816317733e842d2d8/app/models/linehaul.rb).

Since we'll be hitting the API now, I also made sure that the page isn't valid until any outstanding PPM estimate requests have returned.

## Code Review Verification Steps

* [ ] All tests pass.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Request review from a member of a different team.
* [ ] (TRIAL) Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/156899536) for this change

### CODE REVIEW

File: pkg/handlers/personally_procured_move_estimate_test.go [hunk 1]
```diff
@@ -46,6 +46,44 @@ func (suite *HandlerSuite) TestShowPPMEstimateHandler() {
...
+	context := NewHandlerContext(suite.db, suite.logger)
+	context.SetPlanner(route.NewTestingPlanner(1693))
+	showHandler := ShowPPMEstimateHandler(context)
+	showResponse := showHandler.Handle(params)
+
+	// Then: Expect a 200 status code
+	okResponse := showResponse.(*ppmop.ShowPPMEstimateOK)
+	cost := okResponse.Payload
+
+	// And: Returned SIT cost to be as expected
```
Comment (akostibas) [inline pkg/handlers/personally_procured_move_estimate_test.go]: 🍝

File: pkg/rateengine/scenario_test.go [hunk 2]
```diff
@@ -59,6 +58,5 @@ func (suite *RateEngineSuite) Test_Scenario2() {
 	suite.Equal(unit.Cents(430147), cost.LinehaulChargeTotal)
 	suite.Equal(unit.Cents(12103), cost.OriginServiceFee)
 	suite.Equal(unit.Cents(11187), cost.DestinationServiceFee)
-	suite.Equal(unit.Cents(183620), cost.FullPackUnpackFee)
-	suite.Equal(unit.Cents(637057), cost.GCC) // Existing system has only 6,370.56
+	suite.Equal(unit.Cents(637056), cost.GCC)
```
Comment (akostibas) [inline pkg/rateengine/scenario_test.go]: @jim - I made a change to how we were handling Pack & Unpack rates. Can you check out this commit and see if it makes sense?
https://github.com/transcom/mymove/pull/590/commits/6748a1e42be80a88f618ffbac117b1fddf2ca362

Also, it fixed the off-by-one-penny issue we had!

**Comment by akostibas** [comment]
@jim I'm good with the code being in the Rate Engine. I think it makes sense, because we want all consumers of the rate engine to get the same results. If the API returns one result, and the billing system returns another, that would be a problem.

File: pkg/rateengine/scenario_test.go (hunk 2 above)
Comment (jim) [reply pkg/rateengine/scenario_test.go]: @akostibas Looks OK to me.

**Pull request merged**